src/dialogs/more-info-dialog.html

```diff
@@ -86,112 +85,73 @@ <h2>
+        computed: '_computeStateObj(hass)',
+        observer: '_stateObjChanged',
       },
 
-      showHistoryComponent: {
+      _dialogOpen: {
         type: Boolean,
         value: false,
-        computed: 'computeShowHistoryComponent(hasHistoryComponent, stateObj)',
+        observer: '_dialogOpenChanged',
       },
 
-      dialogOpen: {
-        type: Boolean,
-        value: false,
-        observer: 'dialogOpenChanged',
-      },
+      _dialogElement: Object,
+      _registryInfo: Object,
 
-      delayedDialogOpen: {
-        type: Boolean,
-        value: false,
-      },
-
-      cacheConfig: {
-        type: Object,
-        value: {
-          refresh: 60,
-          cacheKey: null,
-          hoursToShow: 24,
-        },
+      _page: {
+        type: String,
+        value: null,
       },
     };
   }
 
-  connectedCallback() {
-    super.connectedCallback();
-    this.$.scrollable.dialogElement = this.$.dialog;
+  ready() {
+    super.ready();
+    this._dialogElement = this.$.dialog;
+    this.addEventListener('more-info-page', (ev) => { this._page = ev.detail.page; });
   }
 
-  computeDomain(stateObj) {
+  _computeDomain(stateObj) {
     return stateObj ? window.hassUtil.computeDomain(stateObj) : '';
   }
 
-  computeStateObj(hass) {
+  _computeStateObj(hass) {
     return hass.states[hass.moreInfoEntityId] || null;
   }
 
-  /**
-   * We depend on a delayed dialogOpen value to tell the chart component
-   * that the data is there. Otherwise the chart component will render
-   * before the dialog is attached to the screen and is unable to determine
-   * graph size resulting in scroll bars.
-   */
-  computeIsLoadingHistoryData(delayedDialogOpen, stateHistoryLoading) {
-    return !delayedDialogOpen || stateHistoryLoading;
-  }
-
-  computeHasHistoryComponent(hass) {
-    return window.hassUtil.isComponentLoaded(hass, 'history');
-  }
-
-  computeShowHistoryComponent(hasHistoryComponent, stateObj) {
-    return this.hasHistoryComponent && stateObj &&
-      window.hassUtil.DOMAINS_WITH_NO_HISTORY
-        .indexOf(window.hassUtil.computeDomain(stateObj)) === -1;
-  }
-
-  stateObjChanged(newVal) {
+  _stateObjChanged(newVal, oldVal) {
     if (!newVal) {
-      this.dialogOpen = false;
+      this._dialogOpen = false;
+      this._page = null;
+      this._registryInfo = null;
       return;
     }
 
-    window.setTimeout(() => {
-      // allow dialog to render content before showing it so it is
-      // positioned correctly.
-      this.dialogOpen = true;
-    }, 10);
-    if (this.cacheConfig.cacheKey !== `more_info.${newVal.entity_id}`) {
-      this.cacheConfig = Object.assign(
-        {}, this.cacheConfig,
-        { cacheKey: `more_info.${newVal.entity_id}` }
-      );
+    if (window.hassUtil.isComponentLoaded(this.hass, 'config.entity_registry') &&
+        (!oldVal || oldVal.entity_id !== newVal.entity_id)) {
+      this.hass.callApi('get', `config/entity_registry/${newVal.entity_id}`)
```

Comment:
For unsupported entity this causes 404 which is printed to the console.
Since this is not actually an error I think the BE should return 200 or another non-error code and the FE should infer `this._registryInfo=false` from the response. WDYT?

Reply:
how about a `204 - No Content`
"The server successfully processed the request and is not returning any content."

Reply:
If you think of the API independently of the frontend, returning a 404 makes sense.

Reply:
Can also just 'catch' the 404 in the frontend, and not let it throw an error. Doesn't really matter. But I think an expected 404 should not give a javascript error at least.

Reply:
Yeah, I agree with that. Let's see what I can cook up.

Reply:
Hmm, looks like we already handle the `error` case of the promise and this is just standard Chrome behavior for XMLHttpRequests that result in a 4XX or 5xx error

Reply:
404 is something that shouldn't happen. So the registry API should either respond with a nonerror code, or provide an API to list all available entities so users don't have to guess.

Reply:
The problem with keeping a list of supported entities around is that the list can go stale.

Reply:
The client can call list immediately before calling get (latency for double call).
Or 'list' can provide full info on all entities (waste of traffic)

Thus I think a non-error response would be best.